token/program-2022/src/extension/confidential_transfer_fee/mod.rs

```diff
@@ -42,6 +51,17 @@ impl Extension for ConfidentialTransferFeeConfig {
     const TYPE: ExtensionType = ExtensionType::ConfidentialTransferFeeConfig;
 }
 
+impl ConfidentialTransferFeeConfig {
+    /// Return the account information needed to construct a `WithdrawWithheldTokensFromMint`
+    /// instruction.
+    #[cfg(not(target_os = "solana"))]
+    pub fn withheld_tokens_info(&self) -> WithheldTokensInfo {
+        WithheldTokensInfo {
+            withheld_amount: self.withheld_amount,
+        }
+    }
+}
```

Comment:
this seems a bit backwards -- how about having a function on `WithheldTokensInfo` that takes `&ConfidentialTransferFeeConfig`? That way you can keep the `#[cfg(not(target_os = "solana"))]` bits all in `account_info.rs`

Reply:
Yeah that is actually a good idea. Since `WithheldTokensInfo` just has one field `withheld_amount`, I think we can just work with the existing constructor `new(withheld_amount: EncryptedWithheldAmount) -> Self` rather than taking in the entire confidential transfer fee config.

But the suggestion appiles for the account info types in the regular confidential transfer extension, so I will apply the suggestion there.

Reply:
https://github.com/solana-labs/solana-program-library/pull/4982